cfme/middleware/__init__.py

```diff
@@ -53,3 +68,63 @@ def download(extension):
         download_btn("Download as {}".format(extensions_mapping[extension]))
     except:
         raise ValueError("Unknown extention. check the extentions_mapping")
+
+
+def add_deployment(filename, runtime_name=None, enable_deploy=True, cancel=False):
```

Comment:
Wouldn't it make more sense to put this as a class method on a `Deployment` class? Then it could navigate to correct place, add the deployment and return an instance of `Deployment` class that will be further usable.